exonum/src/helpers/fabric/details.rs

```diff
@@ -392,6 +406,47 @@ impl GenerateNodeConfig {
 
         (external_address, listen_address)
     }
+
+    /// * `key` - expects EXONUM_SERVICE_PASS or EXONUM_CONSENSUS_PASS.
+    fn get_passphrase(context: &Context, key: &str) -> Vec<u8> {
+        context
+            .get_flag_occurrences(NO_PASSWORD)
+            .map(|_| "".to_string())
+            .or_else(|| env::var(key).ok())
+            .or_else(|| {
+                let stdin = io::stdin();
+                let stdin_locked = stdin.lock();
+                Self::prompt_passphrase(stdin_locked, io::stdout()).ok()
+            }).unwrap()
+            .into_bytes()
+    }
+
+    fn prompt_passphrase<R: BufRead, W: Write>(mut reader: R, mut writer: W) -> io::Result<String> {
+        loop {
+            write!(&mut writer, "Enter passphrase (empty for no passphrase): ").unwrap();
+            let password = rpassword::read_password_with_reader(Some(&mut reader))?;
+            write!(&mut writer, "Enter same passphrase again: ").unwrap();
+            if password == rpassword::read_password_with_reader(Some(&mut reader))? {
+                return Ok(password);
+            }
+            writeln!(&mut writer, "Passphrases do not match.  Try again.").unwrap();
+        }
+    }
+
+    fn create_secret_key_file(
+        secret_key_path: impl AsRef<Path>,
+        passphrase: impl AsRef<[u8]>,
```

Comment:
I'd suggest calling `get_passphrase` from `create_secret_key_file` directly instead of passing the passphrase in. These is not need to use a passphrase outside of this function, and the function itself is an internal utility rather than general-purpose public function. This should be a nicer API for where it's actually used.

I'm not sure whether passing EXONUM_SERVICE_PASS here will read better though...

Reply:
Yeah.. I didn't want to pass variable name to this function, that why I kept it this way. Really dunno which way is really better

Reply:
I think since this function is used in "testnet" where there is no passphrase it can be left as is.
I changed scoping to address your comment in new PR.